### PR TITLE
Support for --kustomize-force-common-label CLI flag

### LINKED ISSUE #4649: Support Kustomize commonLabels --force option
# Summary

ArgoCD should support overriding Kustomize common labels when they are already present in the target repository/path declarative manifests.

# Motivation

ArgoCD already supports setting Kustomize commonLabels imperatively through the CLI, however there is one use case where Kustomize will fail when trying to overwrite existing commonLabels keys present in the declarative manifests. Kustomize exposes a `--force` flag to allow for commonLabels overwriting. 

# Proposal

ArgoCD should support Kustomize commonLabels --force option through the CLI via an opt-in flag `--kustomize-force-common-label`.

### PR DESCRIPTION
**Pull request by oboukili**
Closes #4649

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [x] Does this PR require documentation updates?
* [x] I've updated documentation as required by this PR.
* [x] Optional. My organization is added to USERS.md.
* [x] I've signed the CLA and my build is green ([troubleshooting builds](https://argoproj.github.io/argo-cd/developer-guide/ci/)).

### CODE REVIEW

**Comment by oboukili** [comment]
I would gladly appreciate some help regarding `make codegen` not working on my workstation.

**Comment by oboukili** [comment]
I added the (hopefully) relevant `make codegen` logs in this [gist](https://gist.github.com/oboukili/afa7042969324ac0247969ca06bf0caa), thanks for your help!

File: pkg/apis/application/v1alpha1/types.go [hunk 1]
```diff
@@ -315,6 +315,9 @@ type ApplicationSourceKustomize struct {
 	Images KustomizeImages `json:"images,omitempty" protobuf:"bytes,3,opt,name=images"`
 	// CommonLabels adds additional kustomize commonLabels
 	CommonLabels map[string]string `json:"commonLabels,omitempty" protobuf:"bytes,4,opt,name=commonLabels"`
+	// ForceCommonLabels force overwrites commonLabel directive if it
+	// already exists within the Kustomize application manifests
+	ForceCommonLabels bool `protobuf:"varint,6,opt,name=forceCommonLabels"`
```
Comment (jannfis) [inline pkg/apis/application/v1alpha1/types.go:320]: This should be moved to the bottom of the struct to keep order with the # of the field in the protobuf message.

The new field also lacks the `json` annotation, which is why `gen-crd-spec` panics.

Comment (oboukili) [reply pkg/apis/application/v1alpha1/types.go]: Thanks, that worked!

**Comment by jannfis** [comment]
Hey @oboukili, apologies for the delay. I wanted to merge #4613 before coming back to this one. 

Can you please rebase your branch and adapt to the changes made in #4613? Maybe also incorporate force flag for the common annotations introduced in #4613?

**Comment by josiah-wolf-oberholtzer** [comment]
Any chance of this being revisited?

**Comment by alexmt** [comment]
Closing since https://github.com/argoproj/argo-cd/pull/6443 implemented the same changes

**Pull request closed without merging**